pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,220 @@ def func(df):
+        additional column with the value_counts.
+        By default, rows that contain any NA values are omitted from
+        the result. By default, the result will be in descending order so that the
+        first element of each group is the most frequently-occurring row.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({
+        ...    'gender': ['male', 'male', 'female', 'male', 'female', 'male'],
+        ...    'education': ['low', 'medium', 'high', 'low', 'high', 'low'],
+        ...    'country': ['US', 'FR', 'US', 'FR', 'FR', 'FR']
+        ... })
+
+        >>> df
+            gender 	education 	country
+        0 	male 	low 	    US
+        1 	male 	medium 	    FR
+        2 	female 	high 	    US
+        3 	male 	low 	    FR
+        4 	female 	high 	    FR
+        5 	male 	low 	    FR
+
+        >>> df.groupby("gender").value_counts(normalize=True)
+        gender  education  country
+        female  high       FR         0.50
+                           US         0.50
+        male    low        FR         0.50
+                           US         0.25
+                medium     FR         0.25
+        Name: count, dtype: float64
+        """
+        RESULT_NAME = "count"
+        if self.axis == 1:
+            raise NotImplementedError(
+                "DataFrameGroupBy.value_counts only handles axis=0"
+            )
+
+        with self._group_selection_context():
+            df = self.obj
+
+            # Check for index rather than column grouping
+            index_grouping = self.grouper.groupings[0].name is None
+
+            # Try to find column names
+            if index_grouping:
+                keys: Any = []
+                remaining_columns = self._selected_obj.columns
+            elif isinstance(self._selected_obj, Series):
+                keys = [grouping.name for grouping in self.grouper.groupings]
+                remaining_columns = [self._selected_obj.name]
+            else:
+                if isinstance(self.keys, ops.BaseGrouper):
+                    keys = [grouping.name for grouping in self.keys.groupings]
+                elif isinstance(self.keys, str):
+                    keys = [self.keys]
+                else:
+                    keys = self.keys
+
+                remaining_columns = [
+                    key for key in self._selected_obj.columns if key not in keys
+                ]
+
+            if subset is not None:
+                remaining_columns = [key for key in subset if key not in keys]
+
+            if dropna:
+                df = df.dropna(subset=remaining_columns, axis="index", how="any")
+
+            # Add the remaining_column keys to the main grouper
+            main_grouper, _, _ = get_grouper(
+                df,
+                key=self.keys,
+                axis=self.axis,
+                level=self.level,
+                sort=self.sort,
+                mutated=self.mutated,
+                dropna=dropna,
```

Comment:
I think this should be self.dropna. Users should have the ability to control whether nulls are dropped in the groupers and whether nulls are dropped in the remaining_columns independently. Also, this distinction should be spelled out in the docstring.

Reply:
Done